repos/system_upgrade/el7toel8/libraries/utils.py

```diff
@@ -168,3 +169,47 @@ def read_file(path):
     """
     with open(path, 'r') as f:
         return f.read()
+
+
+def _blank_fn(self, *args, **kwargs):
+    return None
+
+
+def skip_actor_execution_if(*conditions):
```

Comment:
I think that function (or decorator) like that should be something what is implemented inside the framework. Better to say, that we discussed existence of a method for Actor class like `is_executable` (I do not remember the name, @vinzenz ?). Instead of this, I prefer just to add the condition into the process function of the actor now as it is used on several places already to not complicate it, like:
```
def process():
    if architecture.matches_architecture(architecture.ARCH_S390X):
        return
```

Another thing is, that if we decide that the functionality will be part of leapp framework, we will require to use it instead of this decorator and in such a case, people will have additional work to do (as we deprecate that). WDYT @vinzenz?

Reply:
I was thinking about the place to introduce this decorator. I think it shouldn’t be in the Leapp framework. It looks like serves specific needs of IPU only

Reply:
> I was thinking about the place to introduce this decorator. I think it shouldn’t be in the Leapp framework. It looks like serves specific needs of IPU only

I tend to disagree with you here. There is nothing IPU specific about being able to skip actors.

Reply:
OK, then I think the easiest way to solve as proposed by @pirat89 .

Reply:
I will close the MR and do this in a new one